src/Microsoft.Azure.WebJobs.Host/Exceptions/FunctionAbortedException.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Exception thrown when a job function invocation is aborted.
+    /// </summary>
+    public class FunctionAbortedException : FunctionTimeoutException
```

Comment:
What does "abort" mean? From this hierarchy, it looks like you're saying all aborts are due to timeouts? Then why do we need a new exception type

Reply:
Well, it's also breaking for anyone catching a FunctionTimeoutException currently if you start throwing a new derived type that doesn't actually represent a timeout, right? I think this "is-a" relationship is wrong.

Reply:
Updated the naming and definition to:

FunctionAbortTimeoutException - Exception thrown for function invocations that are being aborted due to another concurrently running function timing out.
 
The hierarchy makes much more sense with this naming and logic.